dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/SmsMessageSender.java

```diff
@@ -292,7 +286,12 @@ private void handleResponse(OutboundMessageResponse status, OutboundSms sms) {
       sms.setStatus(OutboundSmsStatus.FAILED);
     }
 
-    outboundSmsService.save(sms);
+    try {
```

Comment:
Who is calling this method?
Why the user is not present?
This doesn't seem to be running in another thread, if I am wrong, who is starting the thread?

Reply:
https://github.com/dhis2/dhis2-core/blob/fd2b4211e66c501e1360bdc6bae7edb3341b2a4a/dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/message/DefaultMessageService.java#L491